### PR TITLE
add contributing guidelines

### LINKED ISSUE #598: Add Community guidelines
Add clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

For [JOSS Review] (https://github.com/openjournals/joss-reviews/issues/6089)

### PR DESCRIPTION
**Pull request by rcannood**
## Describe your changes

Updates to the documentation and templates in the Git repo (#598, PR #600):

  - Add contributing guidelines.

  - Add issue templates.

  - Reworked the pull request template.

## Issue ticket number and link
Closes #598

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality
  - [ ] Major changes
  - [x] Minor changes
  - [ ] Bug fixes

- [x] Proposed changes are described in the CHANGELOG.md

- [ ] Relevant unit tests have been added

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+* Clone your fork to your local machine so you can edit the files. Make sure to check out the `develop` branch, since this is the branch where all development happens.
+
+  ```bash
+  git clone https://github.com/your-username/viash.git
+  cd viash
+  git checkout develop
+  ```
+
+* Create a new branch for your changes. It's best to keep your changes separate from the `develop` branch.
```
Comment (Grifs) [inline CONTRIBUTING.md:40]:
```suggestion
* Create a new branch for your changes. It's best to keep your changes separate from the `develop` branch.

```bash
git branch -b your-branch-name
```
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  ```bash
+  sbt 'testOnly -- -l io.viash.DockerTest -l io.viash.NextflowTest'
+  ```
+
+* Documentation: Update the documentation if your changes require it. This includes both in-code documentation and external documentation like README.md.
+
+## Submitting your changes
+
+* Update the Changelog: Add a new entry to the [CHANGELOG.md](CHANGELOG.md) file that describes your changes. Follow the existing format for entries.
+
```
Comment (Grifs) [inline CONTRIBUTING.md:65]:
```suggestion
* Stage any changed or added files.

```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+* Commit Your Changes: Once you're happy with your changes, commit them to your branch.
+
+  ```sh
+  git commit -m "Add a brief description of your changes"
```
Comment (Grifs) [inline CONTRIBUTING.md:69]: Should we be encouraging single line commit messages? It's generally bad form to do this.

**Review by Grifs — COMMENTED** [review]
Looking good! Added some minor suggestions.

File: CONTRIBUTING.md (hunk 3 above)
Comment (rcannood) [reply CONTRIBUTING.md:76]: Not entirely sure what you mean by this; could you elaborate?

Comment (rcannood) [reply CONTRIBUTING.md:83]: Something like this?

```suggestion
* Commit your changes: Once you're happy with your changes, commit them to your branch. A commit message should consist of a short summary of the changes, followed by an empty line and a more detailed description. For trivial changes, the more detailed description can be omitted.

  ```sh
  git commit
```

Comment (Grifs) [reply CONTRIBUTING.md]: Sounds good to me!

**Pull request merged**